### PR TITLE
nats-streaming image should be created with a non-root user

### PR DESCRIPTION
**Pull request by andriisoldatenko**
nats-streaming image should be created with a non-root user

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
practically LGTM:
```
docker run -it r sh
/ $ id
uid=1000(nats-streaming) gid=1000(nats-streaming) groups=1000(nats-streaming)
/ $
➜  nats-streaming git:(master) ✗ docker run -it r
[1] 2021/07/16 16:01:05.428539 [INF] STREAM: Starting nats-streaming-server[test-cluster] version 0.22.0
[1] 2021/07/16 16:01:05.428625 [INF] STREAM: ServerID: fl8AziVVGDlAcEDi5P7z4M
[1] 2021/07/16 16:01:05.428639 [INF] STREAM: Go version: go1.16.4
[1] 2021/07/16 16:01:05.428653 [INF] STREAM: Git commit: [528588f]
[1] 2021/07/16 16:01:05.429638 [INF] Starting nats-server
[1] 2021/07/16 16:01:05.429690 [INF]   Version:  2.2.6
[1] 2021/07/16 16:01:05.429704 [INF]   Git:      [cf433ae]
[1] 2021/07/16 16:01:05.429718 [INF]   Name:     NBA2JAW4X3RR3LWPKHDKURP6OD3VKEC4LW7XRGNS4GX7X6B52T5KWHAR
[1] 2021/07/16 16:01:05.429732 [INF]   ID:       NBA2JAW4X3RR3LWPKHDKURP6OD3VKEC4LW7XRGNS4GX7X6B52T5KWHAR
[1] 2021/07/16 16:01:05.433251 [INF] Starting http monitor on 0.0.0.0:8222
[1] 2021/07/16 16:01:05.434188 [INF] Listening for client connections on 0.0.0.0:4222
[1] 2021/07/16 16:01:05.434569 [INF] Server is ready
[1] 2021/07/16 16:01:05.459151 [INF] STREAM: Recovering the state...
[1] 2021/07/16 16:01:05.459192 [INF] STREAM: No recovered state
[1] 2021/07/16 16:01:05.460219 [INF] STREAM: Message store is MEMORY
[1] 2021/07/16 16:01:05.460261 [INF] STREAM: ---------- Store Limits ----------
[1] 2021/07/16 16:01:05.460304 [INF] STREAM: Channels:                  100 *
[1] 2021/07/16 16:01:05.460320 [INF] STREAM: --------- Channels Limits --------
[1] 2021/07/16 16:01:05.460334 [INF] STREAM:   Subscriptions:          1000 *
[1] 2021/07/16 16:01:05.460374 [INF] STREAM:   Messages     :       1000000 *
[1] 2021/07/16 16:01:05.460389 [INF] STREAM:   Bytes        :     976.56 MB *
[1] 2021/07/16 16:01:05.460432 [INF] STREAM:   Age          :     unlimited *
[1] 2021/07/16 16:01:05.460474 [INF] STREAM:   Inactivity   :     unlimited *
[1] 2021/07/16 16:01:05.460511 [INF] STREAM: ----------------------------------
[1] 2021/07/16 16:01:05.460546 [INF] STREAM: Streaming Server is ready
^Z^C[1] 2021/07/16 16:01:08.207124 [INF] STREAM: Shutting down.
[1] 2021/07/16 16:01:08.207697 [INF] Initiating Shutdown...
[1] 2021/07/16 16:01:08.207946 [INF] Server Exiting..
```

File: nats-streaming/Dockerfile [hunk 1]
```diff
@@ -13,11 +13,47 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# https://hub.docker.com/_/nats-streaming
-FROM nats-streaming:0.21.2-alpine3.13
+FROM alpine:3.13
```
Comment (vishwas-astro) [inline nats-streaming/Dockerfile:16]: do we want to change to 3.14 here?

Comment (andriisoldatenko) [reply nats-streaming/Dockerfile:16]: we can try

**Review by danielhoherd — APPROVED** [review]

**Comment by danielhoherd** [comment]
@andriisoldatenko should this be merged now or are you doing additional testing?

**Review by vishwas-astro — APPROVED** [review]

**Review by vishwas-astro — APPROVED** [review]

**Pull request merged**